secrets/fake/fake.go

```diff
@@ -0,0 +1,40 @@
+package fake
+
+import (
+	"errors"
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+var (
+	// ErrNotImplemented default secrets in OSD
+	ErrNotImplemented = errors.New("Not Implemented")
+)
+
+type Fake struct {
+	secrets.SecretManager
+}
+
+// New returns a new Fake secret implementation
+func New() *Fake {
+	return &Fake{}
+}
+
+func (f *Fake) SecretLogin(secretType int, secretConfig map[string]string) error {
+	return ErrNotImplemented
+}
+
+func (f *Fake) SetClusterSecretKey(secretKey string, override bool) error {
+	return ErrNotImplemented
+}
+
+func (f *Fake) CheckSecretLogin() error {
+	return ErrNotImplemented
+}
+
+func (f *Fake) SetSecret(secretKey string, secretValue string) error {
```

Comment:
@adityadani, currently we use a map of key-value pairs because a some secrets stores can accept multiple key value pairs. This makes it more flexible to store any kind of data. Should we keep this interface generic as well?

Reply:
Yes. Lets use secretValue as an interface{}